test/VaultTest.t.sol

```diff
@@ -0,0 +1,343 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import "lib/forge-std/src/Test.sol";
+import "lib/forge-std/src/console2.sol";
+
+import "test/VaultMock.sol";
+import "test/USDC.sol";
+import "test/WETH.sol";
+import "test/MockV3Aggregator.sol";
+using stdStorage for StdStorage;
+
+
+abstract contract StateZero is Test {
+    USDC public usdc;
+    WETH9 public weth;
+    MockV3Aggregator public priceFeed;
+
+    VaultMock public vault;
+    address user;
+    address deployer;
+
+    event Deposit(address indexed collateralAsset, address indexed user, uint collateralAmount);  
+    event Borrow(address indexed debtAsset, address indexed user, uint debtAmount); 
+    event Repay(address indexed debtAsset, address indexed user, uint debtAmount);
+    event Withdraw(address indexed collateralAsset, address indexed user, uint collateralAmount);  
+    event Liquidation(address indexed collateralAsset, address indexed debtAsset, address indexed user, uint debtToCover, uint liquidatedCollateralAmount);
+
+
+    function setUp() public virtual {
+        //vm.chainId(4);
+        usdc = new USDC();
+        vm.label(address(usdc), "usdc contract");
+
+        weth = new WETH9();
+        vm.label(address(weth), "weth contract");
+
+        priceFeed = new MockV3Aggregator(18, 1e18);
+        vm.label(address(priceFeed), "priceFeed contract");
+
+        deployer = 0xb4c79daB8f259C7Aee6E5b2Aa729821864227e84;
+        vm.label(deployer, "deployer");
+
+        user = address(1);
+        vm.label(user, "user");
+        
+        // collateralLevel@80% = 0.8 = 8e17
+        vault = new VaultMock(address(usdc), address(weth), address(priceFeed));
+        vm.label(address(vault), "vault contract");
+        
+        //usdc.mint(address(vault), 10000*10**18);
+        //......accessing stdstore instance........\\
+        stdstore    
+        .target(address(usdc))
+        .sig(usdc.balanceOf.selector)         //select balanceOf mapping
+        .with_key(address(vault))            //set mapping key balanceOf(address(vault))
+        .checked_write(100000 * 1e18);      //data to be written to the storage slot -> balanceOf(address(vault)) = 10000*10**18
```

Comment:
how do u feel about using this storage manipulation instead of adding a 'mint' fn to the mock?

Reply:
Well with using storage manipulation, it is slightly divorced from how a real-life implementation would occur (an example would be not taking into consideration the gas fees required for minting the initial underlying asset on deployment)

As a general statement, given this approach utilizes cheats, it could potentially allow us to successfully circumvent scenarios in which we should have failed for one reason or another. Perhaps, when approval was not set before a transfer, but we simulate the end result via manipulation - this might create a vuln or skipped step in the workflow that we might not realize. 

At least this is what occurs to me; could you please share your thoughts as well?

Reply:
Yeah we could control external contracts on forks. But Yield already has mock w mint so Id prob just use mint when i can.

Reply:
So the reason for this was because the second point of the peripheral goal states to use foundry cheats instead of the mint function. 

In the V1 version I used mint, and in V2 cheats to distinguish. Here, I opted to use cheats in the interest of hitting of the required objectives.

Reply:
Yeah I know.  I was just curious how it went.

Reply:
Ahh okay, I thought for a moment you meant that I should change it.

But yes, point taken regarding using mint as the standard approach, and keep cheats to only necessary situations least one gets blinded-sided by the very act of using cheats.